frontends/ol-components/src/components/Input/Input.tsx

```diff
@@ -72,56 +81,63 @@ const Input = styled(InputBase)(({
 }) => {
   return [
     baseInputStyles(theme),
-    size === "medium" && {
-      "& .MuiInputBase-input": {
-        ...theme.typography.body2,
+    (size === "small" || size === "medium") && {
+      ...theme.typography.body2,
+    },
+    (size === "large" || size === "hero") && {
+      ".remixicon": {
+        width: "24px",
+        height: "24px",
       },
+      ...theme.typography.body1,
+      [theme.breakpoints.down("sm")]: {
+        ".remixicon": {
+          width: "20px",
+          height: "20px",
+        },
+        "& .MuiInputBase-input": {
+          ...theme.typography.body3,
+        },
+      },
+    },
+    size === "medium" && {
       paddingLeft: "12px",
       paddingRight: "12px",
-      borderRadius: "4px",
-      "&.MuiInputBase-adornedStart": {
-        paddingLeft: `${12 - buttonPadding.medium}px`,
-      },
-      "&.MuiInputBase-adornedEnd": {
-        paddingRight: `${12 - buttonPadding.medium}px`,
-      },
     },
+    size === "small" &&
+      !multiline && {
+        height: "32px",
+      },
     size === "medium" &&
       !multiline && {
         height: "40px",
       },
-    size === "hero" && {
-      "& .MuiInputBase-input": {
-        ...theme.typography.body1,
-      },
-      paddingLeft: "16px",
-      paddingRight: "16px",
-      borderRadius: "8px",
-      "&.MuiInputBase-adornedStart": {
-        paddingLeft: `${16 - buttonPadding.hero}px`,
-      },
-      "&.MuiInputBase-adornedEnd": {
-        paddingRight: `${16 - buttonPadding.hero}px`,
-      },
-      [theme.breakpoints.down("sm")]: {
-        "& .MuiInputBase-input": {
-          ...theme.typography.body3,
-        },
-        "&.MuiInputBase-adornedStart": {
-          paddingLeft: `${12 - buttonPadding.heroMobile}px`,
-        },
-        "&.MuiInputBase-adornedEnd": {
-          paddingRight: `${12 - buttonPadding.heroMobile}px`,
+    size === "large" &&
+      !multiline && {
+        height: "48px",
+        [theme.breakpoints.down("sm")]: {
+          height: "40px",
         },
```

Comment:
Removing the button padding certainly simplifies stuff 👍 Width is better.

I do worry that we will have a hard time keeping the mobile styles in sync. E.g., if someone updates the `medium` style, will they also remember to update large + breakpoint small?

I do think it would be better to put all the size-related styles into a `sizeStyles`, and re-use that at the smaller breakpoint (like what we do for buttons).

I think this does it: https://github.com/mitodl/mit-open/compare/cg/search-box-updates-2...cg/search-box-updates-2-cc-suggestion#diff-11bf431b91ecb2c9d080ba7edb0f8f9518c9bed7b5fe9cf6154ecb1a835d5ac5L125

**Note:** In that diff, I made `responsive` optional and defaults to undefined ~ false. We could default it to "true" (which this PR effectively does) but I think we should consult Steve about that.